praw/models/reddit/emoji.py

```diff
@@ -89,6 +93,58 @@ def delete(self):
         )
         self._reddit.request("DELETE", url)
 
+    def update(
+        self,
+        mod_flair_only: Optional[bool] = None,
+        post_flair_allowed: Optional[bool] = None,
+        user_flair_allowed: Optional[bool] = None,
+    ):
+        """Update the permissions of an emoji in this subreddit.
+
+        :param mod_flair_only: (boolean) Indicate whether the emoji is
+            restricted to mod use only. Respects pre-existing settings if not
+            provided.
+        :param post_flair_allowed: (boolean) Indicate whether the emoji may
+            appear in post flair. Respects pre-existing settings if not
+            provided.
+        :param user_flair_allowed: (boolean) Indicate whether the emoji may
+            appear in user flair. Respects pre-existing settings if not
+            provided.
+
+        .. note:: In order to retain pre-existing values for those that are not
+           explicitly passed, a network request is issued. To  avoid that
+           network request, explicitly provide all values.
+
+        To restrict the emoji ``test`` in subreddit ``wowemoji`` to mod use
+        only, try:
+
+        .. code-block:: python
+
+           reddit.subreddit("wowemoji").emoji["test"].update(mod_flair_only=True)
+
+        """
+        locals_reference = locals()
+        mapping = {
+            attribute: locals_reference[attribute]
+            for attribute in (
+                "mod_flair_only",
+                "post_flair_allowed",
+                "user_flair_allowed",
+            )
+        }
+        if all(value is None for value in mapping.values()):
+            raise TypeError("At least one attribute must be provided")
```

Comment:
Is it necessary to fail if no changes are made? It can just work, but no changes are made, since there were no changes.

Reply:
I see why you put that in. This can be resolved.